src/query.ts

```diff
@@ -126,12 +167,15 @@ export class Query {
     const filters: Facet = this.params[filterType] as Facet;
 
     if (filters[filterName]) {
-      const index: number = (filters[filterName] as Array<unknown>).indexOf(value);
-
-      if (index !== -1) {
-        (filters[filterName] as Array<unknown>).splice(index, 1);
-        this.params[filterType] = filters;
-        this.params.page = 1;
+      if (!isPlainObject(value)) {
```

Comment:
Why don't you check if it's an array instead?

Reply:
If you use a `Set()` to store filter values you can call `mySet.delete(value)` and it won't break if the value is not there, so a loop will do the job. Maybe: `value.forEach(Set.prototype.delete.bind(mySet))`.

Reply:
I thought that `delete`ing a non-existing value from an Array did nothing too. And my tests seem to prove that.

Reply:
What I mean is that using Map() (as the value of this.params.filters) and Set() (as the value of a concrete filter) this function could be (for terms only):

```js
public removeFilter(filterName, value, filterType = 'filter') {
  // get the filters as a Set() from the Map() of filters
  const filters = this.params[filterType].get(filterName);
  ((typeof value === 'string') ? [value] : value).forEach(
    Set.prototype.delete.bind(filters)
  );
}
```

Now I understand the check for a plain object, you're dealing with ranges being passed as parameter (but doing nothing yet? you should remove the filter entirely in that case). And I'm not sure if your implementation takes care of removing a value passed as string, only as an array of strings…

Reply:
In this case: `removeFilter('brand')` the filter should be removed entirely.